### ISSUE TITLE
Create new draft: server error response on additional properties validation

### ISSUE THREAD

**Issue by audrium** [inveniosoftware/invenio-rdm-records#496]
Package version (if known): <version>

## Describe the bug
If one is using REST API for a draft creation and makes an error in some field name, 500 server error response is returned.

## Steps to Reproduce
1. Get a personal access token from InvenioRDM UI
2. Execute the following by pasting your access token into Authorization header
```python
curl -X POST -w "%{http_code}" \
    -H "Content-Type: application/json" \
    -H "Authorization: Bearer XD09UMM5R9Jli3LqWsyGeY6iBlErTZHohGG9n8999DUUdejUVJkFgz7dNWVY" \
    -d '{"access":{"record":"public","files":"public"},"metadata":{"publication_date":"2021-04-07","publisher":"CERN","rights":[{"description":"The Creative Commons Attribution license allows re-distribution and re-use of a licensed work on the condition that the creator is appropriately credited.","id":"cc-by-4.0","link":"https://creativecommons.org/licenses/by/4.0/legalcode","title":"Creative Commons Attribution 4.0 International"}],"creators":[{"person_or_org":{"type":"personal","given_name":"Doe","family_name":"John","name":"John, Doe"}}],"title":"API Draft Test","additional":"field","resource_type":{"type":"publication","subtype":"publication-annotationcollection"}}}' \
    -k https://127.0.0.1:5000/api/records
```
3. Note that `"additional":"field"` was added to the metadata object
4. It returns:
```python
...
  File "/home/amecioni/.virtualenvs/my-site-Q4AKPVMh/lib/python3.8/site-packages/invenio_records/ext.py", line 35, in validate
    return validate(
  File "/home/amecioni/.virtualenvs/my-site-Q4AKPVMh/lib/python3.8/site-packages/jsonschema/validators.py", line 934, in validate
    raise error
jsonschema.exceptions.ValidationError: Additional properties are not allowed ('additional' was unexpected)

Failed validating 'additionalProperties' in schema['properties']['metadata']:
    {'additionalProperties': False,
     'description': 'Resource metadata.',
...
```

## Expected behavior

400 Bad request error with some explanation what went wrong

**Comment by lnielsen** [comment]
Additional context

1) If we hit a ``jsonschema.exceptions.ValidationError`` it's correct that an 500 internal server error is produced - this is an internal issue to the server as this should never ever happen.

2) In the above case, the additional field should never have been passed on to the record. Instead, it should have raised a validation error in the Marshmallow service schema.

The likely culprit is this [``unknown = INCLUDE``](https://github.com/inveniosoftware/invenio-rdm-records/blob/master/invenio_rdm_records/services/schemas/metadata.py#L455).

However, there's lots of places with ``unknown = EXCLUDE`` - I would instead expect that most places should default to unknown = RAISE`` so that it's an error to send extra data except when explicitly allowed.

This task needs to go through all schemas in Invenio-RDM-Records and ensure only correct data can be submitted.

There's lots of test cases in https://github.com/inveniosoftware/invenio-rdm-records/tree/master/tests/services/schemas that can be enriched.

Basically run a ``git grep EXCLUDE`` and ``git grep INCLUDE`` to check for all possible places where there might be issues with the schema.
